src/main/scala/io/dronekit/cloud/ShellCommandFlow.scala

```diff
@@ -0,0 +1,215 @@
+            log.info("Completing stage")
+            completeStage()
+          }
+          else {
+            if (isAvailable(out)) {
+              val finalPush = outputQueue.descendingIterator().mkString("\n")
+              log.info(s"Pushing final value and completing")
+              push(out, finalPush)
+              completeStage()
+            }
+            else {
+              // Wait for onPull
+              log.info("waiting for onPull to complete stage")
+            }
+          }
+        }
+
+        private def outputAsyncInput(data: Output): Unit = data match {
+          case StandardOutput =>
+            pushAndPull()
+          case ErrorOutput(msg) =>
+            log.error(msg)
+            errorQueue.putFirst(msg)
+          case StandardOutputFinished =>
+            log.debug("StandardOutputFinished")
+            outputClosed.set(true)
+            pushAndComplete()
+
+        }
+
+        setHandler(in, new InHandler{
+          override def onPush: Unit = {
+            val elem = grab(in)
+            if (!errorQueue.isEmpty) {
+              // What is the right thing to do if there is stderr output?
+              errorQueue.clear()
+            }
+
+            inputQueue.putFirst(elem)
+            pushAndPull()
+          }
+
+          override def onUpstreamFinish: Unit = {
+            upstreamFinished.set(true)
+            // This un-blocks processInput allowing it to exit
+            log.debug("onUpstreamFinish")
+            inputQueue.putFirst(ByteString())
+          }
+
+        })
+
+        setHandler(out, new OutHandler {
+          override def onPull(): Unit = {
+            log.debug("onPull")
+            if (outputClosed.get) {
+              pushAndComplete()
+            }
+            else
+              pushAndPull()
+          }
+        })
+
+
+
+        /**
+          * Executed in a separate thread, this method is responsible for
+          * writing the data from the Akka Stream onPush to the process's
+          * InputStream (which is an OutputStream for us)
+          * @param in OutputStream to write to, which is Input for the process
+          */
+        private def processInput(in: OutputStream): Unit = {
+          while (!upstreamFinished.get && !errorAbort.get) {
+            // Block until there is data available
+            val data = inputQueue.takeLast().toArray[Byte]
+            in.write(data)
+          }
+          // Make sure we wrote everything to the input stream
+          while (!inputQueue.isEmpty && !errorAbort.get) {
+            in.write(inputQueue.takeLast().toArray[Byte])
+          }
+          try {
+            in.close()
+          } catch {
+            case ex: IOException if ex.getMessage == "Stream closed" =>
+              log.error(s"Ignoring $ex in processInput while closing")
+            case ex: IOException if ex.getMessage == "Broken pipe" =>
+              log.error(s"Ignoring $ex in processInput while closing")
```

Comment:
does this work? `case ex: IOException if (ex.getMessage == "Broken pipe" || ex.getMessage == "Stream closed")`